libraries/AC_AttitudeControl/AC_PosControl.cpp

```diff
@@ -299,9 +323,11 @@ void AC_PosControl::get_stopping_point_z(Vector3f& stopping_point) const
 /// init_takeoff - initialises target altitude if we are taking off
 void AC_PosControl::init_takeoff()
 {
-    const Vector3f& curr_pos = _inav.get_position();
+    Vector3f curr_pos;
+    _ahrs.get_relative_position_NED_origin(curr_pos);
```

Comment:
No, it shouldn't. Baro is reset every time you arm, which means it should have the same altitude above home, but not necessarily above origin. Home can also be changed and then baro wouldn't match it either, but it was done that way unfortunately.